compiler/qsc_eval/src/lib.rs

```diff
@@ -334,9 +323,13 @@ impl Env {
     }
 
     fn leave_scope(&mut self) {
-        self.0
-            .pop()
-            .expect("scope should be entered first before leaving");
+        // Only pop the scope if there is more than one scope in the stack,
+        // because the global/top-level scope cannot be exited.
```

Comment:
How could this happen? Are the pushes and pops not balanced?

Reply:
The "global" scope used by the interpreter when running top level statements is not really a frame and is never actually pushed. So top-level statements are a kind of weird exception that this logic handles.

Reply:
What if we just push a scope in the very beginning?

Reply:
We sort of do... `Env::default` implemented down on line 376-381 (side note: why is it all the way down there? I'm going to move it up to make it easier to find...) always creates a new environment with an intial global scope. That seemed easier than forcing callers to always have to use `let mut env = Env::default;` followed by `env.push_scope(0);`. But more importantly, because a user can write Q# `return` as a top-level statement, we need to make sure that doesn't accidentally trigger a scope pop that causes all the global variables to dissapper. So scope 0 really is a special, permanent scope that can never be left.